pkg/handler/handler_test.go

```diff
@@ -116,6 +115,84 @@ func TestRegisterOpenAPIVersionedService(t *testing.T) {
 	}
 }
 
+var updatedSwagger = []byte(`{
+  "swagger": "2.0",
+  "info": {
+   "title": "Kubernetes",
+   "version": "v1.12.0"
+  }}`)
+
+func getBodyOrDie(server *httptest.Server) string {
+	req, err := http.NewRequest("GET", server.URL+"/openapi/v2", nil)
+	if err != nil {
+		panic(fmt.Errorf("Unexpected error in creating new request: %v", err))
+	}
+
+	req.Header.Add("Accept", "application/json")
+	resp, err := server.Client().Do(req)
+	if err != nil {
+		panic(fmt.Errorf("Unexpected error in serving HTTP request: %v", err))
+	}
+
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		panic(fmt.Errorf("Unexpected error in reading response body: %v", err))
+	}
+	return string(body)
+}
+
+func normalizeSwaggerOrDie(j []byte) []byte {
+	var s spec.Swagger
+	err := s.UnmarshalJSON(j)
+	if err != nil {
+		panic(fmt.Errorf("Unexpected error in unmarshalling SwaggerJSON: %v", err))
+	}
+	rj, err := json.Marshal(s)
+	if err != nil {
+		panic(fmt.Errorf("Unexpected error in preparing returnedJSON: %v", err))
+	}
+	return rj
+}
+
+func TestUpdateSpecLazy(t *testing.T) {
+	returnedJSON := normalizeSwaggerOrDie(returnedSwagger)
+	var s spec.Swagger
+	err := s.UnmarshalJSON(returnedJSON)
+	if err != nil {
+		t.Errorf("Unexpected error in unmarshalling SwaggerJSON: %v", err)
+	}
+
+	mux := http.NewServeMux()
+	o := NewOpenAPIService(&s)
```

Comment:
Do we need a test for NewOpenAPIServiceLazy?

Reply:
We could but it's not really doing much :-) I'll see if I can add one.

Reply:
NewOpenAPIService actually is basically the same thing as NewOpenAPIServiceLazy so, I don't see a lot of value.